src/renderer/components/+config-maps/config-map-details.tsx

```diff
@@ -102,15 +102,18 @@ class NonInjectedConfigMapDetails extends React.Component<ConfigMapDetailsProps
                 data.map(([name, value]) => (
                   <div key={name} className="data">
                     <div className="name">{name}</div>
-                    <div className="flex gaps align-flex-start">
-                      <Input
-                        multiLine
-                        theme="round-black"
-                        className="box grow"
-                        value={value}
-                        onChange={v => this.data.set(name, v)}
-                      />
-                    </div>
+                    <MonacoEditor
+                      id={`config-map-data-${name}`}
+                      style={{
+                        resize: "vertical",
+                        overflow: "hidden",
+                        border: "1px solid var(--borderFaintColor)",
+                        borderRadius: "4px",
+                      }}
+                      value={value || ""}
```

Comment:
_nit:_ probably better would be to setup default value while extracting fields, e.g. `data.map(([name, value = ""])`